cpu/cc2538/periph/gpio.c

```diff
@@ -26,19 +26,27 @@
 #include "periph/gpio.h"
 
 #define GPIO_MASK           (0xfffff000)
-#define PORTNUM_MASK        (0x00007000)
+#define PORTNUM_MASK        (0x00003000)
 #define PORTNUM_SHIFT       (12U)
 #define PIN_MASK            (0x00000007)
 #define MODE_NOTSUP         (0xff)
 
 static inline cc2538_gpio_t *gpio(gpio_t pin)
 {
-    return (cc2538_gpio_t *)(pin & GPIO_MASK);
+    if(((uint32_t)pin &GPIO_MASK) == 0){
```

Comment:
Changing `PORTNUM_MASK` alone does not work unfortunately, as there is different use of both encodings. While the new one uses the upper bits of the encoding for the port address and the lower bits for the pin number, the old encoding style, as in the pin names, holds no information on which address to access and had to be calculated. So the usage of both is different. A more permanent fix would be to change `cc2538-gpio.h` and make in compatible with this implementation.